bahmni_stock/models/stock_move_line.py

```diff
@@ -0,0 +1,89 @@
+
+from datetime import datetime
+from dateutil import tz
+from collections import Counter, defaultdict
+
+from odoo import models, fields, api
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT as DTF
+
+from odoo.exceptions import UserError, ValidationError, AccessError, RedirectWarning
+
+class StockMoveLine(models.Model):
+    _inherit = 'stock.move.line'     
+    
+    sale_price = fields.Float(string="Sale Price")
+    mrp = fields.Float(string="MRP")
+    cost_price = fields.Float(string="Cost Price")
+    
+    
+    
+    @api.constrains('mrp')
+    def _check_fields_values(self):
+        for rec in self:
+            if rec.mrp > 0.00:        
+                if rec.sale_price > rec.mrp:
+                    raise ValidationError('Sales Price should not be greater than MRP Rate.')
+            else:
+                pass
+                
+    @api.onchange('cost_price')
+    def _check_cost_price_values(self):        
+        if self.cost_price > 0.00:                
+            markup_table_line = self.env['price.markup.table'].search([('lower_price', '<', self.cost_price),
+                                                                       '|', ('higher_price', '>=', self.cost_price),
+                                                                       ('higher_price', '=', 0)],limit=1)
+            if markup_table_line:
+                self.sale_price = self.cost_price + (self.cost_price * markup_table_line.markup_percentage / 100)
+        
+        else:
+            pass
+    
+    @api.model
+    def default_get(self, fields):
+        res = super(StockMoveLine, self).default_get(fields)
+        cost_value = 0.00
+        move_ids = self.env['stock.move'].search([('picking_id', '=', res.get('picking_id')),('product_id', '=', res.get('product_id'))],limit=1)
+        if move_ids:           
+            cost_value = move_ids.purchase_line_id.price_unit + (move_ids.purchase_line_id.price_tax / move_ids.purchase_line_id.product_qty)
+            if cost_value > 0.00:                
+                markup_table_line = self.env['price.markup.table'].search([('lower_price', '<', cost_value),
+                                                                           '|', ('higher_price', '>=', cost_value),
+                                                                           ('higher_price', '=', 0)],limit=1)
+                if markup_table_line:
+                    res.update({'cost_price': cost_value,'sale_price': cost_value + (cost_value * markup_table_line.markup_percentage / 100)})
+            
+            else:
+                pass
+        return res
+        
+    def _create_and_assign_production_lot(self):
```

Comment:
@karthikeyansp91 Where is this function definition invoked from? I don't see any references in the current code

Reply:
@mohan-13 , I have written custom logic to inherit the code for the base module's function, including cost_price, sale_price, mrp, and expiration_date. So, On that time value get and write it.